compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/JvmDefaultArgumentStubGenerator.kt

```diff
@@ -47,4 +47,24 @@ class JvmDefaultArgumentStubGenerator(override val context: JvmBackendContext) :
     }
 
     override fun defaultArgumentStubVisibility(function: IrFunction) = function.getJvmVisibilityOfDefaultArgumentStub()
+
+    override fun needsSuperCallHandlerCheck(irFunction: IrSimpleFunction): Boolean =
+        !irFunction.isTopLevelDeclaration && (irFunction.parent as? IrClass)?.isFinalClass != true
+
+    override fun IrBlockBodyBuilder.generateSuperCallHandlerCheck(
+        irFunction: IrFunction,
+        newIrFunction: IrFunction,
+        handlerDeclaration: IrValueDeclaration
+    ) {
+        +irIfThen(
+            context.irBuiltIns.unitType,
+            irNot(irEqualsNull(irGet(handlerDeclaration))),
+            irThrow(irCall(this@JvmDefaultArgumentStubGenerator.context.ir.symbols.ThrowUnsupportOperationExceptionClass).apply {
+                putValueArgument(
+                    0,
+                    irString("Super calls with default arguments not supported in this target, function: ${irFunction.name.identifier}")
```

Comment:
Please use `asString()` instead of `identifier`

Reply:
Thanks, don't know what happened :) The toString is equivalent, is that fine by you?

Reply:
We specifically prefer to use `Name.asString()` to differentiate between "meaningful" usages of the name contents vs "debug info" usages which traditionally use `toString`.

And `Name.identifier` is undesirable since it throws exception on special names, which IMHO is not needed here (even though I can't think of an example where this could fire right now).

Reply:
Got it, thanks! Updated.